targets/mbedk64f/source/jerry_extapi.cpp

```diff
@@ -0,0 +1,167 @@
+/* Copyright 2014-2015 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerry-core/jerry.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "jerry_extapi.h"
+#include "native_mbedk64f.h"
+
+
+#ifndef MAX
+#define MAX(A,B) ((A)>(B)?(A):(B))
+#endif
+
+
+//-----------------------------------------------------------------------------
+
+#define __UNSED__ __attribute__((unused))
+
+#define DELCARE_HANDLER(NAME) \
+static bool \
+NAME ## _handler (const jerry_api_object_t * function_obj_p __UNSED__, \
+                  const jerry_api_value_t *  this_p __UNSED__, \
+                  jerry_api_value_t *        ret_val_p __UNSED__, \
+                  const jerry_api_value_t    args_p[], \
+                  const jerry_api_length_t   args_cnt)
+
+#define REGISTER_HANDLER(NAME) \
+  register_native_function ( # NAME, NAME ## _handler)
+
+
+//-----------------------------------------------------------------------------
+
+DELCARE_HANDLER(assert)
+{
+  if (args_cnt == 1
+      && args_p[0].type == JERRY_API_DATA_TYPE_BOOLEAN
+      && args_p[0].v_bool == true)
+  {
+    printf (">> Jerry assert true\r\n");
+    return true;
+  }
+  printf ("ERROR: Script assertion failed\n");
+  exit (JERRY_STANDALONE_EXIT_CODE_FAIL);
+  return false;
+}
+
+
+DELCARE_HANDLER(print)
+{
+  jerry_api_length_t cc;
+
+  if (args_cnt)
+  {
+    printf(">> print(%d) :", (int)args_cnt);
+    for (cc=0; cc<args_cnt; cc++)
+    {
+      if (args_p[cc].type == JERRY_API_DATA_TYPE_STRING && args_p[cc].v_string)
+      {
+        static char buffer[128];
+        int length, maxlength;
+        length = -jerry_api_string_to_char_buffer (args_p[0].v_string, NULL, 0);
+        maxlength  = MAX(length, 126);
```

Comment:
If the length is bigger than 128 then the max value will be that number and then we'll be copying more than the size of the buffer into the array, is that really correct? Shouldn't this be MIN instead?